src/query/api/v1/handler/prometheus/remote/match.go

```diff
@@ -65,37 +65,30 @@ func (h *PromSeriesMatchHandler) ServeHTTP(w http.ResponseWriter, r *http.Reques
 	w.Header().Set("Content-Type", "application/json")
 	w.Header().Set("Access-Control-Allow-Origin", "*")
 
-	query, err := prometheus.ParseSeriesMatchQuery(r, h.tagOptions)
+	queries, err := prometheus.ParseSeriesMatchQuery(r, h.tagOptions)
 	if err != nil {
 		logger.Error("unable to parse series match values to query", zap.Error(err))
 		xhttp.Error(w, err, http.StatusBadRequest)
 		return
 	}
 
 	opts := storage.NewFetchOptions()
-	matchers := query.TagMatchers
-	results := make([]*storage.CompleteTagsResult, len(matchers))
-	// TODO: parallel execution
-	for i, matcher := range matchers {
-		completeTagsQuery := &storage.CompleteTagsQuery{
-			CompleteNameOnly: false,
-			TagMatchers:      matcher,
-		}
-
-		result, err := h.storage.CompleteTags(ctx, completeTagsQuery, opts)
+	results := make([]models.Metrics, len(queries))
+	for i, query := range queries {
+		result, err := h.storage.SearchSeries(ctx, query, opts)
 		if err != nil {
 			logger.Error("unable to get matched series", zap.Error(err))
 			xhttp.Error(w, err, http.StatusBadRequest)
 			return
 		}
 
-		results[i] = result
+		results[i] = result.Metrics
 	}
 
 	// TODO: Support multiple result types
-	if renderErr := prometheus.RenderSeriesMatchResultsJSON(w, results); renderErr != nil {
-		logger.Error("unable to write matched series", zap.Error(renderErr))
-		xhttp.Error(w, renderErr, http.StatusBadRequest)
+	if err := prometheus.RenderSeriesMatchResultsJSON(w, results, true); err != nil {
```

Comment:
I don't think we should drop any tags from the results unless that's considered default for Prom. Especially if there's no way for a user to control this.

Reply:
Yeah, reluctantly agree... It's a weird tag that's tacked on by prom but will keep it around for the time being

Reply:
👍